server/src/test/java/org/apache/druid/segment/realtime/appenderator/BatchAppenderatorTester.java

```diff
@@ -0,0 +1,774 @@
+        metrics,
+        dataSegmentPusher,
+        objectMapper,
+        indexIO,
+        indexMerger,
+        rowIngestionMeters,
+        new ParseExceptionHandler(rowIngestionMeters, false, Integer.MAX_VALUE, 0),
+        useLegacyBatchProcessing
+    );
+  }
+
+  private long getDefaultMaxBytesInMemory()
+  {
+    return (Runtime.getRuntime().totalMemory()) / 3;
+  }
+
+  public DataSchema getSchema()
+  {
+    return schema;
+  }
+
+  public AppenderatorConfig getTuningConfig()
+  {
+    return tuningConfig;
+  }
+
+  public FireDepartmentMetrics getMetrics()
+  {
+    return metrics;
+  }
+
+  public ObjectMapper getObjectMapper()
+  {
+    return objectMapper;
+  }
+
+  public Appenderator getAppenderator()
+  {
+    return appenderator;
+  }
+
+  public List<DataSegment> getPushedSegments()
+  {
+    return pushedSegments;
+  }
+
+  @Override
+  public void close() throws Exception
+  {
+    appenderator.close();
+    emitter.close();
+    FileUtils.deleteDirectory(tuningConfig.getBasePersistDirectory());
+  }
+
+  private static File createNewBasePersistDirectory()
+  {
+    return FileUtils.createTempDir("druid-batch-persist");
+  }
+
+
+  // copied from druid-indexing testing since it is not accessible from server module,
+  // Cleaned up a little but Leaving mostly as-is since most of the functionality is 
+  // setting defaults when passing null and those defaults are
+  // required for the appenderator to work
+  private static class TestIndexTuningConfig implements AppenderatorConfig
+  {
+    private static final IndexSpec DEFAULT_INDEX_SPEC = new IndexSpec();
+    private static final int DEFAULT_MAX_PENDING_PERSISTS = 0;
+    private static final boolean DEFAULT_GUARANTEE_ROLLUP = false;
+    private static final boolean DEFAULT_REPORT_PARSE_EXCEPTIONS = false;
+    private static final long DEFAULT_PUSH_TIMEOUT = 0;
+
+    private final AppendableIndexSpec appendableIndexSpec;
+    private final int maxRowsInMemory;
+    private final long maxBytesInMemory;
+    private final boolean skipBytesInMemoryOverheadCheck;
+    private final int maxColumnsToMerge;
+
+    // null if all partitionsSpec related params are null. see getDefaultPartitionsSpec() for details.
+    @Nullable
+    private final PartitionsSpec partitionsSpec;
+    private final IndexSpec indexSpec;
+    private final IndexSpec indexSpecForIntermediatePersists;
+    private final File basePersistDirectory;
+    private final int maxPendingPersists;
+
+    private final boolean forceGuaranteedRollup;
+    private final boolean reportParseExceptions;
+    private final long pushTimeout;
+    private final boolean logParseExceptions;
+    private final int maxParseExceptions;
+    private final int maxSavedParseExceptions;
+    private final long awaitSegmentAvailabilityTimeoutMillis;
+
+    @Nullable
+    private final SegmentWriteOutMediumFactory segmentWriteOutMediumFactory;
+
+    @Nullable
+    private static PartitionsSpec getPartitionsSpec(
```

Comment:
PartitionsSpec in this class is not in use any test. Please stop copying the business logic to the test. All tests must pass a proper partitionsSpec if they test partitioning-related behavior unless they verify the default partitionsSpec.

Reply:
Removed